cub/cub/warp/warp_scan.cuh

```diff
@@ -475,14 +475,80 @@ public:
   //! @param[out] inclusive_output
   //!   Calling thread's output item. May be aliased with `input`
   //!
-  //! @param[in] can_op
+  //! @param[in] scan_op
   //!   Binary scan operator
   template <typename ScanOp>
   _CCCL_DEVICE _CCCL_FORCEINLINE void InclusiveScan(T input, T& inclusive_output, ScanOp scan_op)
   {
     InternalWarpScan(temp_storage).InclusiveScan(input, inclusive_output, scan_op);
   }
 
+  //! @rst
+  //! Computes an inclusive prefix scan using the specified binary scan functor across the
+  //! calling warp.
+  //!
+  //! * @smemwarpreuse
+  //!
+  //! Snippet
+  //! +++++++
+  //!
+  //! The code snippet below illustrates four concurrent warp-wide inclusive prefix max scans
+  //! within a block of 128 threads (one per each of the 32-thread warps).
+  //!
+  //! .. code-block:: c++
+  //!
+  //!    #include <cub/cub.cuh>
+  //!
+  //!    __global__ void ExampleKernel(...)
+  //!    {
+  //!        // Specialize WarpScan for type int
+  //!        typedef cub::WarpScan<int> WarpScan;
+  //!
+  //!        // Allocate WarpScan shared memory for 4 warps
+  //!        __shared__ typename WarpScan::TempStorage temp_storage[4];
+  //!
+  //!        // Obtain one input item per thread
+  //!        int thread_data = ...
+  //!
+  //!        // Compute inclusive warp-wide prefix max scans
+  //!        int warp_id = threadIdx.x / 32;
+  //!        WarpScan(temp_storage[warp_id]).InclusiveScan(thread_data,
+  //!                                                      thread_data,
+  //!                                                      INT_MIN,
```

Comment:
suggestion: this initial value might not be the best from the illustration perspective since it doesn't show up in the output. Having something like `1` instead might be helpful.

Reply:
Also, I totally agree with this take. `ExclusiveScan` shares the same problem.